src/app/stream-deck/msg-handlers.ts

```diff
@@ -0,0 +1,311 @@
+      await delay(250);
+    }
+    dispatch(setSearchQuery(msg.search));
+    // if pull items flag is enabled delay a bit to trigger the action
+    if (msg.pullItems) {
+      await delay(500);
+      // use getState to obtain the updated state after search
+      const loadout = itemMoveLoadout(filteredItemsSelector(getState()), store);
+      await dispatch(applyLoadout(store, loadout, { allowUndo: true }));
+      dispatch(setSearchQuery('', true));
+    }
+  };
+}
+
+// TODO move to a shared module
+function randomizeHandler({ msg, state, store }: HandlerArgs<RandomizeAction>): ThunkResult {
+  return async (dispatch) => {
+    const allItems = allItemsSelector(state);
+    const loadout = randomLoadout(
+      store,
+      allItems,
+      msg.weaponsOnly ? (i) => i.bucket?.sort === 'Weapons' : () => true
+    );
+    loadout && (await dispatch(applyLoadout(store, loadout, { allowUndo: true })));
+  };
+}
+
+function collectPostmasterHandler({ store }: HandlerArgs<CollectPostmasterAction>): ThunkResult {
+  return async (dispatch) => dispatch(pullFromPostmaster(store));
+}
+
+// TODO move to a shared module
+function maximizePowerHandler({ state, store }: HandlerArgs<MaxPowerAction>): ThunkResult {
+  return async (dispatch) => {
+    const allItems = allItemsSelector(state);
+    const loadout = maxLightLoadout(allItems, store);
+    return dispatch(applyLoadout(store, loadout, { allowUndo: true }));
+  };
+}
+
+function farmingModeHandler({ state, store }: HandlerArgs<FarmingModeAction>): ThunkResult {
+  return async (dispatch) => {
+    if (state.farming.storeId) {
+      return dispatch(stopFarming());
+    } else {
+      return dispatch(startFarming(store?.id));
+    }
+  };
+}
+
+function selectionHandler({ msg, state }: HandlerArgs<SelectionAction>): ThunkResult {
+  return async (dispatch) => {
+    const selectionType = msg.selection;
+    dispatch(setSearchQuery(''));
+    dispatch(streamDeckWaitSelection(selectionType));
+    // open the related page
+    const path = selectionType === 'loadout' ? 'loadouts' : 'inventory';
+    if (!window.location.pathname.endsWith(path)) {
+      dispatch(setRouterLocation(routeTo(state, path)));
+      // delay a bit the notification show
+      await delay(100);
+    }
+    // show the notification
+    showSelectionNotification(state, selectionType, () => dispatch(streamDeckClearSelection()));
+  };
+}
+
+function equipLoadoutHandler({ msg, state }: HandlerArgs<EquipOrShareLoadoutAction>): ThunkResult {
+  return async (dispatch) => {
+    const loadouts = loadoutsSelector(state);
+    const stores = storesSelector(state);
+    const store = getStore(stores, msg.character);
+    const loadout = loadouts.find((it) => it.id === msg.loadout);
+    if (store && loadout) {
+      return dispatch(applyLoadout(store, loadout, { allowUndo: true }));
+    }
+  };
+}
+
+function shareLoadoutHandler({ msg, state }: HandlerArgs<EquipOrShareLoadoutAction>): ThunkResult {
```

Comment:
This is the only action I can't quite understand - why would someone share a loadout URL to the streamdeck plugin?

Reply:
I don't know.. that was just an idea, nothing needful for sure

Reply:
I'd say let's leave this one out then